std/encoding.d

```diff
@@ -3380,3 +3380,145 @@ version(unittest)
         return "0123456789ABCDEF"[n & 0xF];
     }
 }
+
+import std.typecons : tuple, Tuple;
+import std.conv : to;
+
+/** Definitions of common Byte Order Marks
+*/
+enum BOM {
+    utf32be   = 0,  /// [0x00, 0x00, 0xFE, 0xFF]
+    utf32le   = 1,  /// [0xFF, 0xFE, 0x00, 0x00]
+    utf7      = 2,  /*  [0x2B, 0x2F, 0x76, 0x38]
+                        [0x2B, 0x2F, 0x76, 0x39],
+                        [0x2B, 0x2F, 0x76, 0x2B],
+                        [0x2B, 0x2F, 0x76, 0x2F],
+                        [0x2B, 0x2F, 0x76, 0x38, 0x2D]
+                    */
+    utf1      = 7,  /// [0xF7, 0x64, 0x4C]
+    utfebcdic = 8,  /// [0xDD, 0x73, 0x66, 0x73]
+    scsu      = 9,  /// [0x0E, 0xFE, 0xFF]
+    bocu1     = 10, /// [0xFB, 0xEE, 0x28]
+    gb18030   = 11, /// [0x84, 0x31, 0x95, 0x33]
+    utf8      = 12, /// [0xEF, 0xBB, 0xBF]
+    utf16be   = 13, /// [0xFE, 0xFF]
+    utf16le   = 14, /// [0xFF, 0xFE]
+    invalid   = 15  /// no BOM was found
+}
+
+/** Mapping of a byte sequence to $(B Byte Order Mark (BOM))
+*/
+enum bomEntries = tuple(
+    [
+        BOM.utf32be,
+        BOM.utf32le,
+        BOM.utf7,
+        BOM.utf7,
+        BOM.utf7,
+        BOM.utf7,
+        BOM.utf7,
+        BOM.utf1,
+        BOM.utfebcdic,
+        BOM.scsu,
+        BOM.bocu1,
+        BOM.gb18030,
+        BOM.utf8,
+        BOM.utf16be,
+        BOM.utf16le,
+        BOM.invalid
+    ],
+    tuple(
+        to!(ubyte[])([0x00, 0x00, 0xFE, 0xFF]),
+        to!(ubyte[])([0xFF, 0xFE, 0x00, 0x00]),
+        to!(ubyte[])([0x2B, 0x2F, 0x76, 0x39]),
+        to!(ubyte[])([0x2B, 0x2F, 0x76, 0x2B]),
+        to!(ubyte[])([0x2B, 0x2F, 0x76, 0x2F]),
+        to!(ubyte[])([0x2B, 0x2F, 0x76, 0x38, 0x2D]),
+        to!(ubyte[])([0x2B, 0x2F, 0x76, 0x38]),
```

Comment:
Yeah, what is the point of this? The sequence ending in 2d will match the sequence ending in 38, why check it twice?